minio/rdma.py

```
@@ -188,6 +195,12 @@ def put(self, bucket: str, obj: str,
         size = length if length is not None else inferred
         if size <= 0:
             raise ValueError("length must be > 0 for RDMA put")
+        if size > _CUOBJ_MAX_MEMORY_REG_SIZE:
+            raise ValueError(
+                f"RDMA put size {size} exceeds the cuObject registration limit "
+                f"of {_CUOBJ_MAX_MEMORY_REG_SIZE} bytes (4 GiB); split into "
+                f"parts <= 4 GiB (multipart upload / ranged read)"
+            )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)